src/lib/__tests__/unshorten.test.js

```diff
@@ -0,0 +1,73 @@
+jest.mock('node-fetch');
+
+const unshorten = require('../unshorten');
+const ResolveError = require('../ResolveError');
+
+const url = 'some url';
+
+describe('unshorten', () => {
+  it('should return location in headers if it exists', async () => {
+    const location = 'some location';
+    require('node-fetch').__setupFetch({
+      headers: {
+        get: jest.fn().mockReturnValue(location),
+      },
+    });
+    const expected = location;
+    expect(await unshorten(url)).toBe(expected);
+  });
+
+  it('should return the url itself if location does not exist in headers', async () => {
+    require('node-fetch').__setupFetch({
+      headers: {
+        get: jest.fn(),
+      },
+    });
+    const expected = url;
+    expect(await unshorten(url)).toBe(expected);
+  });
+
+  it('should throw an error if network timeout happened', async () => {
+    require('node-fetch').__setupFetch(undefined, () => {
+      const err = new Error('network timeout at:');
+      err.name = 'FetchError';
+      throw err;
+    });
+    const test = async () => await unshorten(url);
+    await expect(test()).rejects.toThrow(ResolveError);
```

Comment:
Actually `reject` expects a promise, so we can just remove the `test()` thing and reduce to:
```
await expect(unshorten(url)).rejects.toThrow(ResolveError)
```

Ref:
- https://jestjs.io/docs/en/tutorial-async#rejects
- https://jestjs.io/docs/en/expect#rejects